org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/AbstractRegressionTest.java

```diff
@@ -1402,14 +1402,14 @@ protected ClassFileReader getInternalClassFile(String directoryName, String clas
 			}
 			return classFileReader;
 		} finally {
-			removeTempClass(className);
+			AbstractRegressionTest.removeTempClass(className);
 		}
 	}
 
-	protected void checkDisassembledClassFile(String fileName, String className, String expectedOutput) throws Exception {
-		this.checkDisassembledClassFile(fileName, className, expectedOutput, ClassFileBytesDisassembler.DETAILED);
+	protected static void checkDisassembledClassFile(AbstractRegressionTest abstractRegressionTest, String fileName, String className, String expectedOutput) throws Exception {
+		AbstractRegressionTest.checkDisassembledClassFile(fileName, className, expectedOutput, ClassFileBytesDisassembler.DETAILED);
```

Comment:
the prefix "AbstractRegressionTest." feels unnecessary

Reply:
It is syntactically unnecessary, that's true. You could remove it here and it is just a matter of preference whether you want to have this class name for calls of static methods within the same class.
We have actively chosen to implement it this way and also reflected this in test cases for the refactoring (e.g. `testThisInDeclaration`). The decision was guided by the fact that, e.g., IntelliJ does it the same way.
We can, of course, also implement it differently. If you see good reasons to change that, we can open and address an according issue.

Reply:
personally idk, since i probably will never use that refactoring. i don't get the value of it. I had hoped it would be a useful cleanup that would just add "static" wherever "this" is not needed.

Reply:
We have also thought about reusing the refactoring functionality to provide a quick fix based on it that shows up for methods that can be made static without adding a parameter (as they do not rely on `this`). That would a follow-up task. Value of the refactoring of course depends on your use cases. We sometimes have them and probably others have them as well (otherwise competitors like IntelliJ would have probably not have them implemented).